content/docs/rfcs/38/README.md

```diff
@@ -0,0 +1,714 @@
+final opinion on the truth of the proposal.
+
+### Setup Parameters
+
+The node initializes the following integer ratios as constants:
+```
+;;; The following values are constants chosen with justification from experiments
+;;; performed with the adversarial models
+
+;; FIXME: find and justify an empirical confidence threshold 
+confidence_threshold
+  <-- 1  ;; BOGUS:  this should be a function of the network size
+         ;; and the current confidence in the observer majority as 
+         ;; a function of the current round.  With a value of `1`
+         ;; the algorithm will never terminate
+
+;;; constant look ahead for number of rounds we expect to finalize a
+;;; decision.  Could be set dependent on number of nodes 
+;;; visible in the current gossip graph.
+look_ahead 
+  <-- 19
+;;; These need to be justified empirically via modeling and simulation.
+certainty ;; the confidence weighting parameter (aka alpha_1)
+  <-- 4 / 5  
+doubt ;; the lack of confidence weighting parameter (aka alpha_2)
+  <-- 2 / 5 
+
+k_multiplier     ;; neighbor threshold multiplier
+  <-- 2
+
+;;; maximal threshold multiplier, i.e. we will never exceed 
+;;; questioning k_initial * k_multiplier ^ max_k_multiplier_power peers
+max_k_multiplier_power 
+  <-- 4
+    
+;;; Initial numbers of nodes queried in a round
+k_initial 
+  <-- 7
+
+;;; maximum query rounds before termination
+max_rounds
+   <-- 997 ;; TODO justify
+```
+      
+The following variables are needed to keep the state of Glacier:
+
+```
+;; current number of nodes to attempt to query in a round
+k 
+  <-- k_original
+  
+;; total number of votes examined over all rounds
+total_votes 
+   <-- 0 
+;; total number of YES (i.e. positive) votes for the truth of the proposal
+total_positive 
+   <-- 0
+;; the current query round, an integer starting from zero
+round
+  <-- 0
+```
+
+
+###  Query 
+
+A node selects `k` nodes randomly from the current locally known
+complete set of peers in the network.
+
+A query is sent to each neighbor with the node's current `opinion` of
+the proposal.
+
+Each node replies with their current opinion on the proposal.
+
+See [the wire protocol Interoperability section](#wire-protocol) for
+details on the semantics and syntax of the "on the wire"
+representation of this query.
+
+When the query finishes, the node now initializes the following two
+values:
+
+    new_votes 
+      <-- |total vote replies received in this round to the current query|
+    positive_votes 
+      <-- |YES votes received from the query| 
+    
+### Computation
+
+The node runs the `new_votes` and `positive_votes` parameters received
+in the query round through the following algorithm:
+
+    total_votes 
+      +== new_votes
+    total_positive 
+      +== positive_votes
+    confidence 
+      <-- total_votes / (total_votes + look_ahead) 
+    total_evidence 
+      <-- total_positive / total_votes
+    new_evidence 
+      <-- positive_votes / new_votes
```

Comment:
Suppose that a node queried $K$ nodes but  all  $K$  haven't  responded.  In this case   positive_votes=0 and  new_votes=0 and we will get 0/0 division, i.e. new_evidence is undefined. Intuitively when we have no responses we have no evidence, but we need to decide on have to handle this situation.

Reply:
In the implementation we just compute things if we at least 1 vote. https://github.com/logos-co/consensus-research/blob/101cf169e4665e6720ce6ee3c5bc582337b73cd1/consensus/claro/src/claro.rs#L248